src/Tempest/Http/HttpApplication.php

```diff
@@ -16,33 +17,29 @@
 
 final readonly class HttpApplication implements Application
 {
-    public function __construct(
-        private Container $container,
-        private AppConfig $appConfig,
-    ) {
-    }
+    public function __construct(private Container $container) {}
 
-    public static function boot(string $root, ?AppConfig $appConfig = null): self
+    /** @param \Tempest\Core\DiscoveryLocation[] $discoveryLocations */
+    public static function boot(
+        string $root,
+        array $discoveryLocations = [],
+    ): self
     {
-        $container = Tempest::boot($root, $appConfig);
-
-        // TODO: check if we need this
-        $appConfig = $container->get(AppConfig::class);
+        $container = Tempest::boot($root, $discoveryLocations);
```

Comment:
We should probably allow the container to be passed in. If we are going to override core services (like the event bus) we're going to have to have the ability to control what we pass in.

Reply:
In that case people can construct the kernel themselves I think? This `Tempest::boot` method is only here for "the default flow"

Reply:
Yeah, I think we would construct it manually in the entrypoint, maybe?